convert2rhel/checks.py

```diff
@@ -75,81 +93,79 @@ def perform_pre_ponr_checks():
+            "Couldn't check if the current installed Convert2RHEL is the latest version.\n"
+            "repoquery failed with the following output:\n%s" % (raw_output_convert2rhel_versions)
         )
         return
 
-    PKG_NEVR = r"\b(\S+)-(?:([0-9]+):)?(\S+)-(\S+)\b"
     convert2rhel_versions = re.findall(PKG_NEVR, raw_output_convert2rhel_versions, re.MULTILINE)
-    latest_version = ("0", "0.00", "0")
+    logger.debug("Found %s convert2rhel package(s)" % len(convert2rhel_versions))
+    latest_available_version = ("0", "0.00", "0")
 
+    # This loop will determine the latest available convert2rhel version in the yum repo.
+    # It assigns the epoch, version, and release ex: ("0", "0.26", "1.el7") to the latest_available_version variable.
     for package_version in convert2rhel_versions:
-        # rpm.lableCompare(pkg1, pkg2) compare two kernel version strings and return
-        # -1 if str2 is greater then str1, 0 if they are equal, 1 if str1 is greater the str2
-        ver_compare = rpm.labelCompare(package_version[1:], latest_version)
+        # rpm.labelCompare(pkg1, pkg2) compare two package version strings and return
+        # -1 if latest_version is greater than package_version, 0 if they are equal, 1 if package_version is greater than latest_version
+        ver_compare = rpm.labelCompare(package_version[1:], latest_available_version)
         if ver_compare > 0:
-            latest_version = package_version[1:]
-
-    # After the for loop latest_version will have the epoch ,version, and release ex:("0" "0.26" "1.el7") information from convert2rhel yum repo.
-    # the release for latest_verion will be "1.el7" and the relase for convert2rhel_version will be hard coded as "0" below,
-    # therefore when the versions are the same the latest_version's release field will cause it to evaluate as later
-    ver_compare = rpm.labelCompare(("0", convert2rhel_version, "0"), ("0", latest_version[1], "0"))
+            logger.debug(
+                "...convert2rhel version %s is newer than %s, updating latest known version"
+                % (package_version[1:][1], latest_available_version[1])
+            )
+            latest_available_version = package_version[1:]
+
+    # After the for loop, the latest_available_version variable will gain the epoch, version, and release
+    # (e.g. ("0" "0.26" "1.el7")) information from the Convert2RHEL yum repo
+    # when the versions are the same the latest_available_version's release field will cause it to evaluate as a later version.
+    # Therefore we need to hardcode "0" for both the epoch and release below for installed_convert2rhel_version
+    # and latest_available_version respectively, to compare **just** the version field.
+    ver_compare = rpm.labelCompare(("0", installed_convert2rhel_version, "0"), ("0", latest_available_version[1], "0"))
     if ver_compare < 0:
         if "CONVERT2RHEL_UNSUPPORTED_VERSION" in os.environ:
```

Comment:
```suggestion
        if "CONVERT2RHEL_ALLOW_OLDER_VERSION" in os.environ:
```
As we've decided to stop using the "UNSUPPORTED" word, @abadger, @SpyTec, @r0x0d, are you ok with this suggestion?

Reply:
The new name works for me.

Reply:
Works for me too.